File: tutorials/testing/java/book_snippet.java
import java.util.List;
import java.util.Random;
import bridges.connect.Bridges;
import bridges.connect.DataSource;
import bridges.data_src_dependent.GutenbergBook;

/*! A test class that fetches a random book from Gutenberg Book then sends the
	title, authors name, and genre to the console as output.
	There is no visual output to the users BRIDGES gallery after running this code.*/
	
public class book_snippet {
	public static void main(String[] args) throws Exception {

		//create the Bridges object
		Bridges bridges = new Bridges(YOUR_ASSIGNMENT_NUMBER, "YOUR_USER_ID", "YOUR_API_KEY");

		/* Get a List of GutenbergBook objects from Bridges */
    DataSource ds = bridges.getDataSource();
    List<GutenbergBook> mylist = ds.getGutenbergBookMetaData();

		/* Inspect a random GutenbergBook object */
		GutenbergBook book1 = mylist.get((new Random()).nextInt(mylist.size()));
		System.out.println(book1.getTitle());
		System.out.println(book1.getAuthorName());
		System.out.println(book1.getGenres());

	}
}

File: tutorials/testing/python/book_snippet.py
from bridges.bridges import *
from bridges.data_src_dependent.data_source import *
import sys
import random

def main():
    args = sys.argv[1:]

    # create the Bridges object, set credentials
    bridges = Bridges(int(args[0]), args[1], args[2])

    if len(args) > 3:
        bridges.connector.set_server(args[3])

        book_list = get_gutenberg_book_data()

        book1 = book_list[random.randrange(len(book_list))]

        print(book1.title)
        print(book1.name)
        print(book1.genre)

    if __name__ == '__main__':
        main()